pkg/azure/api/mock/mock_test.go

```diff
@@ -79,41 +70,41 @@ func (e *MockSuite) TestMock(c *check.C) {
 	vmInstances.Update("vm2", ipamTypes.InterfaceRevision{
 		Resource: resource.DeepCopy(),
 	})
-	c.Assert(err, check.IsNil)
-	c.Assert(vmInstances.NumInstances(), check.Equals, 1)
+	require.NoError(t, err)
+	require.Equal(t, 1, vmInstances.NumInstances())
 	vmInstances.ForeachInterface("", func(instanceID, interfaceID string, iface ipamTypes.InterfaceRevision) error {
-		c.Assert(instanceID, check.Equals, "vm2")
-		c.Assert(interfaceID, check.Equals, vmIfaceID)
+		require.Equal(t, "vm2", instanceID)
+		require.Equal(t, vmIfaceID, interfaceID)
 		return nil
 	})
 
 }
 
-func (e *MockSuite) TestSetMockError(c *check.C) {
+func TestSetMockError(t *testing.T) {
 	api := NewAPI([]*ipamTypes.Subnet{}, []*ipamTypes.VirtualNetwork{})
-	c.Assert(api, check.Not(check.IsNil))
+	require.NotNil(t, api)
 
 	mockError := errors.New("error")
 
 	api.SetMockError(GetInstances, mockError)
 	_, err := api.GetInstances(context.Background(), ipamTypes.SubnetMap{})
-	c.Assert(err, check.Equals, mockError)
+	require.Equal(t, mockError, err)
 
 	api.SetMockError(GetVpcsAndSubnets, mockError)
 	_, _, err = api.GetVpcsAndSubnets(context.Background())
-	c.Assert(err, check.Equals, mockError)
+	require.Equal(t, mockError, err)
 
 	api.SetMockError(AssignPrivateIpAddressesVMSS, mockError)
 	err = api.AssignPrivateIpAddressesVMSS(context.Background(), "vmss1", "i-1", "s-1", "eth0", 0)
-	c.Assert(err, check.Equals, mockError)
+	require.Equal(t, mockError, err)
```

Comment:
nit: maybe consider `require.ErrorIs` here? I'm not sure if it's exactly the same though. 🤔 
```suggestion
	require.ErrorIs(t, err, mockError)
```

Reply:
ah it's exactly the same due to our mock sestup.